src/compositemark/boxplot.ts

```diff
@@ -285,30 +285,40 @@ function boxParams(spec: GenericUnitSpec<Encoding<string>, BOXPLOT | BoxPlotDef>
   ];
   let postAggregateCalculates: CalculateTransform[] = [];
 
-  if (isMinMax) {
-    aggregate.push({
-      op: 'min',
-      field: continuousAxisChannelDef.field,
-      as: 'lowerWhisker'
-    });
-    aggregate.push({
-      op: 'max',
-      field: continuousAxisChannelDef.field,
-      as: 'upperWhisker'
-    });
-  } else {
+  const preappendWhiskerType: string = isMinMax ? '' : 'minmax_';
+
+  aggregate.push({
+    op: 'min',
+    field: continuousAxisChannelDef.field,
+    as: preappendWhiskerType + 'lowerWhisker'
+  });
+  aggregate.push({
+    op: 'max',
+    field: continuousAxisChannelDef.field,
+    as: preappendWhiskerType + 'upperWhisker'
+  });
+
+  if (!isMinMax) {
     postAggregateCalculates = [
       {
         calculate: 'datum.upperBox - datum.lowerBox',
         as: 'IQR'
       },
       {
         calculate: 'datum.lowerBox - datum.IQR * ' + kIQRScalar,
-        as: 'lowerWhisker'
+        as: 'iqr_lowerWhisker'
       },
       {
         calculate: 'datum.upperBox + datum.IQR * ' + kIQRScalar,
+        as: 'iqr_upperWhisker'
+      },
+      {
+        calculate: 'min(datum.iqr_upperWhisker, datum.minmax_upperWhisker)',
```

Comment:
You can calculate `iqr_lowerWhisker` inline here:

`calculate: 'min(datum.upperBox + datum.IQR * ${kIQRScalar},  datum.minmax_upperWhisker)'`